### PR TITLE
[#11177] Create new user role: maintainer

### LINKED ISSUE #11177: Create new user role: Senior Developer
**Environment**: `V7`

**Description of feature/enhancement**
A new user role, Senior Developer is added to the system to allow the system to recognize which user is a senior developer to TEAMMATES.

**Justification**
This can help senior developers to access certain restrict resources so turnaround time between an error occurs and developers respond to it can reduce.

### PR DESCRIPTION
**Pull request by hhdqirui**
Fixes #11177

**Outline of Solution**
1) Modify `UserInfo.java` to have a `isDeveloper` Boolean field and getter methods for it.
2) Create a static list of `UserId` to “Developer” mapping for prod server instance. The config file that we choose to store this list can be `build.properties`. The format would be a comma separated list. For example: `app.seniorDeveloper.ids = id1000,id1001,id1002,id1003`. For getting this property and converting to a java list, we can use the `split` method.
3) In `teammates/common/util/Config.java`, add a `seniorDevelopers` field with type `List<String>` and methods to load this list from `build.properties`.
4) Add a new method `isUserSeniorDeveloper(String id)` in `UserProvision.java` to check whether the `id` given in the parameter is in the list of senior developer (`Config.seniorDevelopers`). This new method will be called in `getCurrentLoggedInUser()` and set `isDeveloper` of `userInfo` object accordingly.

### CODE REVIEW

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]

File: src/main/java/teammates/logic/api/UserProvision.java [hunk 1]
```diff
@@ -36,6 +37,7 @@ protected UserInfo getCurrentLoggedInUser(UserInfoCookie uic) {
 
         UserInfo userInfo = new UserInfo(uic.getUserId());
         userInfo.isAdmin = uic.isAdmin();
+        userInfo.isSeniorDeveloper = this.isUserSeniorDeveloper(userInfo.getId());
```
[the remark this answers is not in the document]
Comment (hhdqirui) [reply src/main/java/teammates/logic/api/UserProvision.java]: OK. It is fixed.

File: src/main/java/teammates/logic/api/UserProvision.java [hunk 2]
```diff
@@ -59,4 +61,11 @@ public UserInfo getAdminOnlyUser(String userId) {
         return userInfo;
     }
 
+    /**
+     * Checks whether the given id is in the list of senior developers' id.
+     */
+    public boolean isUserSeniorDeveloper(String id) {
+        return Config.SENIOR_DEVELOPERS.contains(id);
```
[the remark this answers is not in the document]
Comment (hhdqirui) [reply src/main/java/teammates/logic/api/UserProvision.java]: Thank you for the advice. I have removed the method.

File: src/main/resources/build.template.properties [hunk 3]
```diff
@@ -126,3 +126,6 @@ app.mailjet.secretkey =
 
 # This is the host URL for the full-text search service used by the system.
 app.search.service.host=http\://localhost\:8983/solr
+
+# User ids of senior developers
+app.seniorDeveloper.ids =
```
[the remark this answers is not in the document]
Comment (hhdqirui) [reply src/main/resources/build.template.properties]: Thank you for the advice. It is fixed.

File: src/main/java/teammates/common/util/Config.java [hunk 4]
```diff
@@ -133,6 +136,7 @@
         SEARCH_SERVICE_HOST = properties.getProperty("app.search.service.host");
         ENABLE_DATASTORE_BACKUP = Boolean.parseBoolean(properties.getProperty("app.enable.datastore.backup", "false"));
         MAINTENANCE = Boolean.parseBoolean(properties.getProperty("app.maintenance", "false"));
+        SENIOR_DEVELOPERS = Arrays.asList(properties.getProperty("app.seniorDeveloper.ids").split(","));
```
[the remark this answers is not in the document]
Comment (hhdqirui) [reply src/main/java/teammates/common/util/Config.java]: It is fixed.

File: src/main/java/teammates/common/util/Config.java [hunk 5]
```diff
@@ -98,6 +98,9 @@
     /** The value of the "app.maintenance" in build.properties file. */
     public static final boolean MAINTENANCE;
 
+    /** The value of the "app.seniorDeveloper.ids" in build.properties file. */
+    public static final List<String> SENIOR_DEVELOPERS;
```
[the remark this answers is not in the document]
Comment (hhdqirui) [reply src/main/java/teammates/common/util/Config.java]: Fixed.

File: src/main/resources/build.template.properties (hunk 3 above)
[the remark this answers is not in the document]
Comment (hhdqirui) [reply src/main/resources/build.template.properties]: Thank you for the advice. It is fixed. Do you also advice to change other variable and method names that contain `seniorDeveloper` to `seniorDev`?

**Comment by rrtheonlyone** [comment]
@hhdqirui please also update the name to "maintainer" as discussed

**Comment by hhdqirui** [comment]
@wkurniawan07 @rrtheonlyone Hi it is ready for review.

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]
There are still mentions of "senior developers" in code.

**Comment by hhdqirui** [comment]
It is all cleared now. No more "senior developer".

**Review by wkurniawan07 — APPROVED** [review]
LGTM

**Pull request merged**